Source/SwiftLintFramework/Rules/LegacyConstantRule.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  LegacyConstantRule.swift
+//  SwiftLint
+//
+//  Created by Aaron McTavish on 12/01/2016.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import SourceKittenFramework
+import CoreGraphics
+
+public struct LegacyConstantRule: CorrectableRule {
+    public static let description = RuleDescription(
+        identifier: "legacy_constant",
+        name: "Legacy Constant",
+        description: "Struct-scoped constants are preferred over legacy global constants.",
+        nonTriggeringExamples: [
+            "CGRect.infinite",
+            "CGPoint.zero",
+            "CGRect.zero",
+            "CGSize.zero",
+            "CGRect.null"
+        ],
+        triggeringExamples: [
+            "↓CGRectInfinite",
+            "↓CGPointZero",
+            "↓CGRectZero",
+            "↓CGSizeZero",
+            "↓CGRectNull"
+        ],
+        corrections: [
+            "CGRectInfinite\n": "CGRect.infinite\n",
+            "CGPointZero\n": "CGPoint.zero\n",
+            "CGRectZero\n": "CGRect.zero\n",
+            "CGSizeZero\n": "CGSize.zero\n",
+            "CGRectNull\n": "CGRect.null\n"
+        ]
+    )
+
+    public func validateFile(file: File) -> [StyleViolation] {
+        let constants = ["CGRectInfinite", "CGPointZero", "CGRectZero", "CGSizeZero",
+            "CGRectNull"]
+
+        let pattern = "\\b(" + constants.joinWithSeparator("|") + ")\\b"
+
+        return file.matchPattern(pattern, withSyntaxKinds: [.Identifier]).map {
+            StyleViolation(ruleDescription: self.dynamicType.description,
+                location: Location(file: file, characterOffset: $0.location))
+        }
+    }
+
+    public func correctFile(file: File) -> [Correction] {
+        let patterns = [
+            "CGRectInfinite": "CGRect.infinite",
+            "CGPointZero": "CGPoint.zero",
+            "CGRectZero": "CGRect.zero",
+            "CGSizeZero": "CGSize.zero",
+            "CGRectNull": "CGRect.null"
+        ]
+
+        let description = self.dynamicType.description
+        var corrections = [Correction]()
+        var contents = file.contents
+
+        for (pattern, template) in patterns {
+            let matches = file.matchPattern(pattern, excludingSyntaxKinds: [.Comment])
```

Comment:
Why not filter on `.Identifier` like you're doing in `validateFile`? As it stands, the string references above would get corrected and that's certainly not what we want.

And here you're scanning the entire file for each pattern, whereas `validateFile` just does it once for all patterns.